extensions/fleet/fleet.go

```diff
@@ -0,0 +1,76 @@
+package fleet
+
+import (
+	"time"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	FleetGitRepoResourceType = "fleet.cattle.io.gitrepo"
+)
+
+// CreateFleetGitRepo is a "helper" functions that takes a rancher client, and the rke2 cluster config as parameters. This function
+// registers a delete cluster fuction with a wait.WatchWait to ensure the cluster is removed cleanly.
```

Comment:
nit, should be `ensure the gitrepo object is removed` instead of `cluster is removed`.